pkg/noderesourcetopology/score.go

```diff
@@ -0,0 +1,117 @@
+package noderesourcetopology
+
+import (
+	"context"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	apiconfig "sigs.k8s.io/scheduler-plugins/pkg/apis/config"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+)
+
+const (
+	defaultWeight = int64(1)
+)
+
+type scoreStrategy func(v1.ResourceList, v1.ResourceList, resourceToWeightMap) int64
+
+// resourceToWeightMap contains resource name and weight.
+type resourceToWeightMap map[v1.ResourceName]int64
+
+// weight return the weight of the resource and defaultWeight if weight not specified
+func (rw *resourceToWeightMap) weight(r v1.ResourceName) int64 {
+	w, ok := (*rw)[r]
+	if !ok {
+		return defaultWeight
+	}
+
+	if w < 1 {
+		return defaultWeight
+	}
+
+	return w
+}
+
+func (tm *TopologyMatch) Score(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
+	// calculates the fraction of requested to capacity per each numa-node.
+	// return the numa-node with the minimal score as the node's total score
+	return tm.score(pod, nodeName)
+}
+
+func (tm *TopologyMatch) ScoreExtensions() framework.ScoreExtensions {
+	return nil
+}
+
+// score will use `scoreStrategy` function to calculate the score.
+func (tm *TopologyMatch) score(pod *v1.Pod, nodeName string) (int64, *framework.Status) {
+	klog.V(5).Infof("Call score for node %v", nodeName)
+	nodeTopology := findNodeTopology(nodeName, &tm.nodeResTopologyPlugin)
+
+	if nodeTopology == nil {
+		return 0, nil
+	}
+
+	klog.V(5).Infof("nodeTopology: %v", nodeTopology)
+	calculateResourceAllocatableRequest(pod, nodeTopology)
+	resources, allocatablePerNUMA := calculateResourceAllocatableRequest(pod, nodeTopology)
+	return scoreForEachNUMANode(resources, allocatablePerNUMA, tm.scoreStrategy, tm.resourceToWeightMap)
```

Comment:
score you calculated here is not applicable for ContainerLevel TM policy, we need to take it into account here.

Reply:
Done